Add Container build files and automation

NeonDaniel opened the pull request:
# Description
Add websocket dockerfile and overlay
Add automation to build and publish websocket Docker image

# Issues
<!-- If this is related to or closes an issue/other PR, please note them here -->

# Other Notes
The design intent here is to build separate Docker containers for different kinds of nodes

NeonDaniel committed:
Add websocket dockerfile and overlay
Add automation to build and publish websocket Docker image

mikejgray reviewed (changes requested):
This should probably not run as root by default

NeonDaniel committed:
Update dependencies to allow newer OVOS packages
Update Websocket Client for compat. with updated OVOS dependencies
Update WS Docker to use Pulse audio
Update Node Docker config to a user with Node permissions
Fix default audio resource file handling
Document Docker

NeonDaniel committed:
Update neon_node bundled configs to valid beta credentials
Add Docker build tests

NeonDaniel committed:
Update voice_client for newer Dinkum listener compat
Add voice Dockerfile and GHA

NeonDaniel commented:
> This should probably not run as root by default

I tried working through this, but ran into permissions issues with the Pulse cookie file that's mounted to the container. Do you know of any examples for how to deal with this?

NeonDaniel committed:
Update shared workflow reference for docker build tests

NeonDaniel commented:
I did a little more troubleshooting today based on [ovos-docker](https://github.com/OpenVoiceOS/ovos-docker/blob/dev/compose/docker-compose.yml#L70) and some back and forth with Claude but still end out with permissions errors.. It appears both are using ALSA plugins, but I'm not able to work around this error with Pulse:
```
ALSA lib pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Connection refused

2025-03-10 22:49:11.111 - neon-node - ovos_microphone_plugin_alsa:_run:110 - ERROR - Failed to open microphone
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/ovos_microphone_plugin_alsa/__init__.py", line 75, in _run
    mic = alsaaudio.PCM(
alsaaudio.ALSAAudioError: Connection refused [default]
```

Open to commits/PR to fix this, but otherwise I suggest keeping these running as `root` for the time being
